README.md

```diff
@@ -16,4 +16,6 @@ Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
 
-The run-time analysis of the merge sort algorithm, which I believe would be O(n log (n)) for it’s worst case runtime. I believe this because it divides the array into smaller parts and then combines them while comparing and rearranging elements. The number of operations for each step is proportional to the array size with O(n log (n)) time complexity. So the worst case bound is $\Theta$ (n log (n))
+
+The run-time analysis of the merge sort algorithm which I believe would be $\ O(n log (n))$ , because it involves dividing the array into smaller parts and then combining them, while comparing and rearranging elements. The division occurs in the first loop, in the outer loop, which iterates $\log_2(n)$ times, where n is the array size, to split the array into smaller segments. The merging process, performed within the inner loop,  ensures the sorted combination of these segments, involving element shifting as needed to maintain order, especially when merging elements from the left and right subarrays. The worst-case time complexity will be $\Theta$ $\(n log (n))$, which the time it takes to sort the array increases with the array size.
```

Comment:
Don't think that this is correct -- have a look at your merge part and see what the complexity is, in particular compared to the original mergesort.